packages/react-dom/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -827,21 +832,35 @@ function pushInput(
           );
         // eslint-disable-next-line-no-fallthrough
         case 'defaultChecked':
-          // Previously "checked" would win but now it's enumeration order dependent.
-          // There's a warning in either case.
-          pushAttribute(target, responseState, 'checked', propValue);
+          defaultChecked = propValue;
           break;
         case 'defaultValue':
-          // Previously "value" would win but now it's enumeration order dependent.
-          // There's a warning in either case.
-          pushAttribute(target, responseState, 'value', propValue);
+          defaultValue = propValue;
+          break;
+        case 'checked':
+          checked = propValue;
+          break;
+        case 'value':
+          value = propValue;
           break;
         default:
           pushAttribute(target, responseState, propKey, propValue);
           break;
       }
     }
   }
+
+  if (checked !== null) {
```

Comment:
Why this instead of `props.checked`? Well because we have to do the loop and check every property name anyway. Where as `props.checked` likely wouldn't be fast pathed since the hidden class of these props are often different. So it would be an unnecessary map look up. However, I do a props.id check below but that's just because I expect that code path to go away.